### PR TITLE
feat: state logs in the exported file with app version and build number

### PR DESCRIPTION
**Pull request by tommasini**
## **Description**
Adding the app versions and build number to the state logs txt file

## **Related issues**

Fixes:

## **Manual testing steps**

1. Go to this page...
2.
3.

## **Screenshots/Recordings**

IOS sim: https://recordit.co/mvaITSTtQC

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've clearly explained what problem this PR is solving and how it is solved.
- [ ] I've linked related issues
- [ ] I've included manual testing steps
- [ ] I've included screenshots/recordings if applicable
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.
- [ ] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
  - [ ] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: app/util/logs/index.test.ts [hunk 1]
```diff
@@ -14,6 +14,8 @@ jest.mock('../../core/Engine', () => ({
 describe('logs :: generateStateLogs', () => {
   it('should generate the state logs correctly without the explicitly deleted controller states', async () => {
     const mockStateInput = {
+      appVersion: 1,
+      buildNumber: 123,
       engine: {
```
Comment (NicolasMassart) [inline app/util/logs/index.test.ts:19]: to make this test more clear about the fact that the version and build infos are not part of the state initially, and remove possible confusion, you can remove values from mock state and add them at generateStateLog call time like you do on the actual code:
```suggestion
    const mockStateInput = {
      engine: {
```

File: app/util/logs/index.test.ts [hunk 2]
```diff
@@ -24,12 +26,15 @@ describe('logs :: generateStateLogs', () => {
       },
     };
     const logs = generateStateLogs(mockStateInput);
```
Comment (NicolasMassart) [inline app/util/logs/index.test.ts:28]:
```suggestion
    const logs = generateStateLogs({
        ...mockStateInput,
        appVersion: 1,
        buildNumber: 123,
      }
    );
```

File: app/util/logs/index.test.ts (hunk 1 above)
Comment (tommasini) [reply app/util/logs/index.test.ts]: Test case added ! Great point!

File: app/util/logs/index.test.ts [hunk 3]
```diff
@@ -24,12 +24,39 @@ describe('logs :: generateStateLogs', () => {
...
+    expect(logs.includes('NftController')).toBe(false);
+    expect(logs.includes('TokensController')).toBe(false);
+    expect(logs.includes('AssetsContractController')).toBe(false);
+    expect(logs.includes('TokenDetectionController')).toBe(false);
+    expect(logs.includes('NftDetectionController')).toBe(false);
+    expect(logs.includes('PhishingController')).toBe(false);
+    expect(logs.includes("vault: 'vault mock'")).toBe(false);
+  });
+
+  it('should generate extra logs if they added to the state object', () => {
```
Comment (NicolasMassart) [inline app/util/logs/index.test.ts:37]:
```suggestion
  it('generates extra logs if values added to the state object parameter', () => {
```

**Review by NicolasMassart — DISMISSED** [review]
Looks good to me.
just a nit about the test name (do not use "should",see guidelines) but if you want to keep it like this as the other test is also using this naming, it's probably fine.

**Review by NicolasMassart — APPROVED** [review]
Looks very good to me.

**Pull request merged**